File: TourApp/api/src/test/java/com/example/api/TourScrappingIntegration.java
package com.example.api;

import org.junit.Test;

import java.io.IOException;

import static org.junit.Assert.*;

public class TourScrappingIntegration {

    @Test
    public void getStagesEndpoint() throws IOException {
        TourScrappingService api = new ServiceFactory().build(TourScrappingService.class);
        if(api.getStages().execute().body()!=null){
            assertFalse(api.getStages().execute().body().isEmpty());
        } else {
            fail("Body is empty and it shouldn't main API endpoint");
        }
    }
}

File: TourApp/api/src/test/java/io/github/pedrofraca/tour/api/TourScrappingIntegration.kt
package io.github.pedrofraca.tour.api

import org.junit.Assert
import org.junit.Test
import java.io.IOException

class TourScrappingIntegration {
    @Throws(IOException::class)
    @Test
    fun stagesEndpoint() {
        val api = ServiceFactory().build(TourScrappingService::class.java)
        val stages = api.stages().blockingGet()
        Assert.assertFalse(stages?.isEmpty() ?: true)
    }

    @Throws(IOException::class)
    @Test
    fun classificationEndpoint() {
        val api = ServiceFactory().build(TourScrappingService::class.java)
        val stages = api.getClasificationForStage("1").blockingGet()
        Assert.assertFalse(stages.general.isEmpty())
    }
}